File: flutter_frontend/lib/widgets/bilingual_field.dart
import 'package:flutter/material.dart';

class BilingualField extends StatelessWidget {
  const BilingualField({
    super.key,
    required this.primaryLabel,
    required this.secondaryLabel,
    required this.child,
    this.helperText,
  });

  final String primaryLabel;
  final String secondaryLabel;
  final Widget child;
  final String? helperText;

  @override
  Widget build(BuildContext context) {
    // Keep the field label on the active language only.
    // 仅保留当前语言字段标题，避免输入控件被双语文案挤压。
    final theme = Theme.of(context);
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          primaryLabel,
          style: theme.textTheme.labelLarge?.copyWith(
            fontWeight: FontWeight.w600,
          ),
        ),
        const SizedBox(height: 8),
        child,
        if (helperText != null && helperText!.trim().isNotEmpty) ...[
          const SizedBox(height: 6),
          Text(
            helperText!,
            style: theme.textTheme.bodySmall?.copyWith(
              color: theme.colorScheme.onSurfaceVariant,
            ),
          ),
        ],
      ],
    );
  }
}

File: frontend/components/bilingual-field.js
window.BilingualField = {
  props: {
    primaryLabel: {
      type: String,
      required: true,
    },
    secondaryLabel: {
      type: String,
      required: true,
    },
    helperText: {
      type: String,
      default: '',
    },
  },
  // Reusable bilingual field shell.
  // Reusable field shell for the active language only.
  // 仅显示当前语言的字段外壳，避免标签区堆叠双语文案。
  template: `
    <div class="identity-field">
      <label class="identity-label">
        <span class="identity-label-main">{{ primaryLabel }}</span>
      </label>
      <slot></slot>
      <div v-if="helperText" class="form-hint">{{ helperText }}</div>
    </div>
  `,
};
